vispy/visuals/filters/mesh.py

```diff
@@ -309,15 +312,20 @@ def __init__(self, shading='flat', light_dir=(10, 5, -5),
                  shininess=100):
         self._shading = shading
         self._light_dir = light_dir
-        # TODO: Accept a single float in [0, 1] instead of a full color for
-        # convenience. Set the color to (1, 1, 1) and the alpha channel to the
-        # float value.
-        self._ambient_light = Color(ambient_light)
-        self._diffuse_light = Color(diffuse_light)
-        self._specular_light = Color(specular_light)
-        self._ambient_coefficient = Color(ambient_coefficient)
-        self._diffuse_coefficient = Color(diffuse_coefficient)
-        self._specular_coefficient = Color(specular_coefficient)
+
+        def _as_rgba(intensity_or_color):
+            if isinstance(intensity_or_color, numbers.Number):
+                intensity = intensity_or_color
+                return Color((1, 1, 1, intensity))
+            color = intensity_or_color
+            return Color(color)
```

Comment:
@djhoese What do you think about this? This wraps the `vispy.color.color_array.Color` class for this specific use case and I'm not sure it fits in `Color` directly because the color of a scalar, e.g. `Color(0.3)`, doesn't make sense. Maybe a new `ColorAndIntensity` class would reflect the intent better. Should I keep it this way here?

Also, I'm thinking of moving `_as_rgba()` in the global scope of this submodule to be able to unit-test it.

Reply:
Could this be replaced with `as_vec4`:

https://github.com/vispy/vispy/blob/acafcb07a0c9947c319e1611d72cf7045670a8e7/vispy/visuals/transforms/_util.py#L50

Also, whatever method you choose you need to use in all the property `.setter`s too, right? I'm not sure this use case is unique enough to require a new class or even a new utility. This seems like something that should already exist in vispy (as_vec4?), but obviously if it doesn't then yeah we need it. There may be a different way of structuring this though too. Would doing something like `Color(as_vec4(intensity_or_color, default=(1, 1, 1, 1)))` work?

Reply:
`as_vec4()` doesn't in this case as it fills the default vec4 from left to right.

Reply:
Yes, it should be in the `.setter`s too. Thanks!

Reply:
@djhoese I've settled for keeping the new `_as_rgba()`. Let me know if you have further thougs/preferences.